tk205/file_io.py

```diff
@@ -1,39 +1,44 @@
 import os
 import json
 import cbor2
+import yaml
 from .xlsx import template, A205XLSXTree
 
 def get_extension(file):
     return os.path.splitext(file)[1]
 
-def load(file):
-    # todo check file exisits
-    ext = get_extension(file).lower()
+def load(input_file_path):
+    ext = get_extension(input_file_path).lower()
     if (ext == '.json'):
-        with open(file, 'r') as input_file:
+        with open(input_file_path, 'r') as input_file:
             return json.load(input_file)
     elif (ext == '.cbor'):
-        with open(file, 'rb') as input_file:
+        with open(input_file_path, 'rb') as input_file:
             return cbor2.load(input_file)
     elif (ext == '.xlsx'):
         tree = A205XLSXTree()
-        return tree.load_workbook(file).get_content()
+        return tree.load_workbook(input_file_path).get_content()
+    elif (ext == '.yaml'):
+        return yaml.load(input_file_path, Loader=yaml.FullLoader)
     else:
         raise Exception(f"Unsupported input \"{ext}\".")
 
-def dump(content, file):
-    # todo check file exisits
-    ext = get_extension(file).lower()
+def dump(content, output_file_path):
+    ext = get_extension(output_file_path).lower()
     if (ext == '.json'):
-        with open(file,'w') as output_file:
+        with open(output_file_path,'w') as output_file:
             json.dump(content, output_file, indent=4)
     elif (ext == '.cbor'):
-        with open(file,'wb') as output_file:
+        with open(output_file_path,'wb') as output_file:
             cbor2.dump(content, output_file)
     elif (ext == '.xlsx'):
         tree = A205XLSXTree()
         tree.load(content)
-        tree.save(file)
+        tree.save(output_file_path)
+    elif (ext == '.yaml'):
+        with open(output_file_path, 'w') as out_file:
+            out_file.write(yaml.dump(content, default_flow_style=False))
```

Comment:
I tried `yaml.dump(content, output_file)` and it works just fine. Is there a reason not to use that form? Do we know what `default_flow_style=False` is doing and why it might be necessary?

Reply:
I think it's ok to use the `yaml.dump(content, output_file)` format. The docs generally use the call the way I am using it, but I don't see any reason why we should use the two file method.

The `default_flow_style=False` is necessary for us, it ensures that nested collections are "dumped" correctly. There is a section about this in the docs for pyyaml, but to summarize briefly, the default flow style prints in block format, not serial. So if we had the data
`a: 1
b: {c: 3, d: 4}`
and we don't set `default_flow_style=False`, then that is exactly what will be written to the file, which is not proper yaml. Setting the `default_flow_style=False` ensures that the output will look like:
`a: 1
b:
  c: 3
  d: 4`
which is what we want to happen. I hope that makes sense. If not, take a look at the "frequently asked" section of the docs [here](https://pyyaml.org/wiki/PyYAMLDocumentation) and it should make more sense

Reply:
Let's remove it as long as it's getting the output we want.